File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    //variable lowest 
    //variable highest
    //variable max
    //iterate through prices
    //return max
    var lowest = Infinity;
    var highest = -Infinity;
    var max = 0;
    for (let price of prices) {
        if (price < lowest) {
            lowest = price;
            highest = -Infinity;
        } else if (price > highest) {
            highest = price;
        }
        
        highest - lowest > max ? max = highest - lowest : null;
    }
    
    return max;
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l, r = 0, 1
        maxP = 0
        while r < len(prices):
            if prices[r] < prices[l]:
                l = r
            maxP = max(prices[r] - prices[l], maxP)
            r += 1
        return maxP